src/ToolCallProgressCards.tsx

```diff
@@ -0,0 +1,42 @@
+import { useSyncExternalStore } from 'react';
+import { ToolCallStreamStore } from './stores/toolCallStreamStore';
+import { OperationRegistry } from './types/chain';
+import { TextStreamStore } from './stores/textStreamStore';
+
+export const ToolCallProgressCards = ({
+  onRendered,
+  toolCallStreamStore,
+  operationRegistry,
+  progressStore,
+  isOperationValidated,
+}: {
+  onRendered: () => void;
+  progressStore: TextStreamStore;
+  toolCallStreamStore: ToolCallStreamStore;
+  operationRegistry: OperationRegistry<unknown>;
+  isOperationValidated: boolean;
+}) => {
+  const toolCallStreams = useSyncExternalStore(
+    toolCallStreamStore.subscribe,
+    toolCallStreamStore.getSnapShot,
+  );
```

Comment:
Does it make sense to subscribe to this store in the `ConversationWrapper` and then just pass down the array of tool call streams?